AdsCommentsServiceImpl

EDyugaeva opened the pull request:
Change in model of Comments (ManyToOne with author and ads) -> change in mapper
AdsCommentsServiceImpl

EDyugaeva committed:
AdsCommentsServiceImpl
Change in model of Comments (ManyToOne with author and ads) -> review in mapper

EDyugaeva committed:
AdsCommentsServiceImpl
Change in model of Comments (ManyToOne with author and ads) -> review in mapper

Corrections in models and mappers
+ ControllerAdvisor

Delegant commented on src/main/java/ru/skypro/homework/models/mappers/AdsMapper.java:
```diff
@@ -1,21 +1,43 @@
+@Mapper(componentModel = "spring")
 public interface AdsMapper {
     AdsMapper INSTANCE = Mappers.getMapper(AdsMapper.class);
 
```
это не нужно, мне кажется

Delegant commented on src/main/java/ru/skypro/homework/repository/AdsCommentsRepository.java:
```diff
@@ -2,8 +2,15 @@
 @Repository
 public interface AdsCommentsRepository extends JpaRepository<Comments, Integer> {
+    Collection<Comments> findCommentsByAds(Ads ads);
+
```
лучше list

Delegant commented on src/main/java/ru/skypro/homework/models/mappers/AdsMapper.java:
```diff
@@ -1,21 +1,43 @@
+    @Mapping(target = "email", source = "author.email")
+    @Mapping(target = "phone", source = "author.phone")
+    @Mapping(target = "image", source = "image.pk")
     FullAdsDto toDullAdsDto(Ads ads);
```
опечатка

Delegant commented on src/main/java/ru/skypro/homework/models/exceptions/ErrorResponse.java:
```diff
@@ -0,0 +1,20 @@
+
+    private String message;
+    private HttpStatus httpStatus;
+
```
👍

Delegant commented on src/main/java/ru/skypro/homework/service/impl/AdsCommentsServiceImpl.java:
```diff
@@ -1,42 +1,122 @@
+    private final UserRepository userRepository;
 
     @Override
     public List<AdsCommentDto> getAdsComments(String ad_pk) {
```
в схеме как то неудачно названы переменные, может заменим на верблюжью нотацию?

EDyugaeva replied:
Поддерживаю, в своих сервисах поменяла

Delegant commented on src/main/java/ru/skypro/homework/service/impl/AdsCommentsServiceImpl.java:
```diff
@@ -1,42 +1,122 @@
+        checkOnExistingAuthor(adsCommentDto);
+
+        Comments comment = CommentsMapper.INSTANCE.toComments(adsCommentDto);
+        Ads ads = adsRepository.findById(ad_pk_int).orElse(new Ads());
```
не очень понял в какой ситуации возможен кейс, когда нужно создать ads для комента, если он не нашелся. Может лучше кидать ошибку тут? Или я не понял как работает?

EDyugaeva replied:
Логично
Дело в том, что findById может выкидывать Optional и из него нужно сделать объект))
можно добавить в репозиторий метод findAdsById, чтобы был чистый Ads

Проверка на существование объявления есть

Delegant commented on src/main/java/ru/skypro/homework/service/impl/AdsCommentsServiceImpl.java:
```diff
@@ -1,42 +1,122 @@
+        Integer ad_pk_int = Integer.parseInt(ad_pk);
+        checkOnExistingAds(ad_pk_int);
+        checkOnExistingComment(id);
+        Comments comments = adsCommentsRepository.findCommentsByPk(id);
```
может лучше возвращать opthional и кидать эксепшен если он пустой, а то так получается два обращения в базу просто ради проверки.

EDyugaeva replied:
Ну тут разные таблицы, так что мне кажется ок

Delegant commented on src/main/java/ru/skypro/homework/service/impl/AdsCommentsServiceImpl.java:
```diff
@@ -1,42 +1,122 @@
+        adsCommentsRepository.deleteById(id);
     }
 
     @Override
```
тут в такой реализации нужно добавить аннотацию Transactional, так как возможен кейс, когда на момент проверки базы адс будет в наличии, но к моменту сохранения коммента его кто то удалит. И тогда сервер вернет 500 статус.

Delegant reviewed (approved):
написал свое мнение в комментариях, в целом можно не править, только коллекцию на лист заменить. Обработка ошибок прикольная, надо добавить в остальные методы тоже.

EDyugaeva committed:
AdsCommentsServiceImpl
Change in model of Comments (ManyToOne with author and ads) -> review in mapper

Corrections in models and mappers
+ ControllerAdvisor

EDyugaeva committed:
AdsCommentsServiceImpl
Change in model of Comments (ManyToOne with author and ads) -> review in mapper

Corrections in models and mappers
+ ControllerAdvisor
